src/formats/tomboy/core.ts

```diff
@@ -0,0 +1,605 @@
+		return text.replace(/([\\`*_\[\]#])/g, '\\$1');
+	}
+
+	/**
+	 * Stream-based formatting analysis for a line of content sections
+	 */
+	private formatLineStream(sections: Array<ContentSection>, isBoldIgnored: boolean, removeStrikethrough: boolean = false): string {
+		let result = '';
+		let activeFormats: Set<string> = new Set();
+		let trailingWhitespacePrevious = '';
+
+		sections.forEach((section, sectionIndex) => {
+			let sectionText = section.text;
+
+			// Check if this section is an internal link (needed early for text processing)
+			const isLink = section.xmlPath.includes('link:internal');
+
+			// Extract and preserve whitespace that should stay outside formatting
+			const { coreText, leadingWhitespace, trailingWhitespace } = this.extractWhitespaceFromText(sectionText);
+
+			// Escape markdown special characters in core text to prevent interference,
+			// but for internal links, use sanitizeFileName to match note title transformations
+			const escapedCoreText = isLink
+				? sanitizeFileName(coreText)
+				: this.escapeMarkdownSpecialChars(coreText);
+
+			// Determine formatting changes for this section's core content
+			const formats: { [key: string]: boolean } = {
+				strikethrough: !removeStrikethrough && section.xmlPath.includes('strikethrough'),
+				highlight: section.xmlPath.includes('highlight'),
+				bold: !isBoldIgnored && section.xmlPath.includes('bold'),
+				italic: section.xmlPath.includes('italic'),
+				monospace: section.xmlPath.includes('monospace'),
+			};
+
+			// Handle formatting transitions for core content
+			const formatChanges = this.calculateFormattingChanges(activeFormats, formats);
+
+			// Apply closing formats before trailing whitespace
+			result += formatChanges.closeTags;
+
+			// Add whitespace (always outside formatting)
+			result += trailingWhitespacePrevious + leadingWhitespace;
+
+			// Apply opening formats before core text
+			result += formatChanges.openTags;
+
+			// Push new spans to stack (after opening tags are applied)
+			if (formatChanges.openingSpans) {
+				formatChanges.openingSpans.forEach(span => this.activeFormatStack.push(span));
+			}
+
+			// Internal links are always assumed to span just 1 section
+			if (isLink) {
+				result += `[[`;
+			}
+
+			// Add the escaped core section text
+			result += escapedCoreText;
+
+			if (isLink) {
+				result += `]]`;
+			}
+
+			// Update active formats
+			activeFormats = new Set(Object.keys(formats).filter(key => formats[key]));
+
+			trailingWhitespacePrevious = trailingWhitespace;
+		});
+
+		// Close any remaining formats from stack (preserves proper ordering)
+		while (this.activeFormatStack.length > 0) {
+			result += this.activeFormatStack.pop()!.closeTag;
+		}
+
+		return result;
+	}
+
+	/**
+	 * Calculate formatting changes between current and new format states using stack-based ordering
+	 */
+	private calculateFormattingChanges(currentFormats: Set<string>, newFormats: { [key: string]: boolean }) {
```

Comment:
Since this is only called in one place, I think it would be easier to reason about if it were moved in-line so that closing tags can modify the result directly. Let me know what you think.